Sources/Hummingbird/Server/RequestContext.swift

```diff
@@ -71,21 +61,21 @@ public struct HBCoreRequestContext: Sendable {
 /// Protocol that all request contexts should conform to. Holds data associated with
 /// a request. Provides context for request processing
 public protocol HBBaseRequestContext: Sendable {
+    associatedtype Decoder: HBRequestDecoder = NullDecoder
+    associatedtype Encoder: HBResponseEncoder = NullEncoder
```

Comment:
Does it make sense to make these primary associated types?

Reply:
I'm not sure that is necessary. When are you going to need to specify the encoder in a parameter type